homeassistant/components/edp_redy.py

```diff
@@ -0,0 +1,132 @@
+"""
+Support for EDP re:dy.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/edp_redy/
+"""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD,
+                                 EVENT_HOMEASSISTANT_START)
+from homeassistant.core import callback
+from homeassistant.helpers import discovery, dispatcher
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_point_in_time
+from homeassistant.util import dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'edp_redy'
+EDP_REDY = "edp_redy"
+DATA_UPDATE_TOPIC = '{0}_data_update'.format(DOMAIN)
+UPDATE_INTERVAL = 30
+
+REQUIREMENTS = ['edp_redy==0.0.1']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the EDP re:dy component."""
+    from edp_redy import EdpRedySession
+
+    session = EdpRedySession(config[DOMAIN][CONF_USERNAME],
```

Comment:
Please update so that the `EdpRedySession` accepts an aiohttp session to reuse. We want all aiohttp clients in home assistant to reuse the same session. Get the aiohttp session from home assistant with the aiohttp helper, as in the previous commit.

Read background in the note here:
https://aiohttp.readthedocs.io/en/stable/client_quickstart.html

Reply:
Thanks, I'll do.
I should also pass the hass event loop for async_timeout, right?
```
with async_timeout.timeout(DEFAULT_TIMEOUT, loop=self._hass.loop):
```

Reply:
Yes, if you need the loop, do that.